pkg/kapp/crdupgradesafety/change_validator.go

```diff
@@ -113,6 +113,98 @@ func RequiredFieldChangeValidation(diff FieldDiff) (bool, error) {
 	return handled(), nil
 }
 
+func MinimumChangeValidation(diff FieldDiff) (bool, error) {
+	handled := func() bool {
+		diff.Old.Minimum = nil
+		diff.New.Minimum = nil
+		return reflect.DeepEqual(diff.Old, diff.New)
+	}
+
+	switch {
+	case diff.Old.Minimum == nil && diff.New.Minimum != nil:
+		m := *diff.New.Minimum
+		return handled(), fmt.Errorf("minimum constraint added when one did not exist previously: %+v", m)
+	case diff.Old.Minimum != nil && diff.New.Minimum != nil:
+		oldMin := *diff.Old.Minimum
+		newMin := *diff.New.Minimum
+		if oldMin < newMin {
```

Comment:
Can switch to the below to avoid creating new variables: 

```suggestion
		if math.Float64bits(*diff.Old.Minimum) < math.Float64bits(*diff.New.Minimum) {
```

Reply:
The new variables are needed. See https://github.com/carvel-dev/kapp/pull/944#discussion_r1585037631 for an explanation

Reply:
That makes sense. I suggested this based on the changes I have for increasing maximums, but I haven't started writing e2e tests yet so I am not aware of the pitfalls. But in the unit tests, I had to create new variables as well. Something like the below: 

```
max := float64(10)
Old: &v1.JSONSchemaProps{
      Maximum: &max,
},
```

But it looks like you are able to set it as `Minimum: pointer.Float64(10)` so I am going to switch to use this instead. 😃